pkg/operatorstatus/operatorstatus.go

```diff
@@ -0,0 +1,124 @@
+package operator
+
+import (
+	"fmt"
+
+	metaapi "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/util/retry"
+
+	osapi "github.com/openshift/cluster-version-operator/pkg/apis/operatorstatus.openshift.io/v1"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+
+	"github.com/openshift/cluster-samples-operator/pkg/apis/samplesoperator/v1alpha1"
+)
+
+func UpdateOperatorStatus(sampleResource *v1alpha1.SamplesResource) error {
```

Comment:
@gabemontero i expect this logic to evolve over time, but hopefully it's a sufficient stake in the ground for now.